discordbot/cogs/auditlogs.py

```diff
@@ -0,0 +1,97 @@
+import discord
+from discord.ext import commands
+
+class AuditLog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+    
+    async def is_admin_or_executive_check(ctx):
+        # Check if the user is an admin
+        if ctx.author.guild_permissions.administrator:
+            return True
+
+        # Check if the user has the 'executive' role
+        return any(role.name.lower() == 'executive' for role in ctx.author.roles)
+
+    @commands.command(name='auditlogs', aliases=['AuditLogs'])
+    @commands.check(is_admin_or_executive_check)
+    async def auditlogs(self, ctx, user: discord.User = None, limit: int = 10):
+        # call with !auditlogs @Username <Number of logs to print>
+        # Example: !auditlogs @John 5
+        if user is None:
+            await ctx.send("Usage: `!auditlogs @Username [Num Logs to Print]`\nExample: `!auditlogs @John 5`")
+            return
+        try:
+            # Fetch the audit logs for the server
+            logs = []
+            async for log in ctx.guild.audit_logs(limit=limit, user=user):
+                logs.append(log)
```

Comment:
in this for loop in order to deal with channels that have been deleted you can set 
`log.target.name = log.[before/after].name`
you can also do something along the lines of
`log.target.type = log.[before/after].type`
and then change all the is instance() checks to target.type == 'type needed'
This would prevent specifying the type of channel but that is less important than having readable logs

Reply:
The problem is log.before and log.extra don't reliably work for deleted objects (atleast I think). The solution we discussed in the call was to have a text channel in which the bot dumps a record any time it performs a command. 

I can change what it currently does to just say a generic message for deleted objects like "@User deleted text channel" instead of the long ID it prints right now.

Reply:
I see fair enough. I suppose this isn't ultimately too important. I think replacing the the id is a good idea to make it seem a little cleaner. Beyond that the rest of your code looks good, so do that and I'll approve.